README_NEW.md

```diff
@@ -0,0 +1,349 @@
+# Go backend for MLflow
+
+In order to increase the performance of the tracking server and the various stores, we propose to rewrite the server and store implementation in Go.
+
+
+# Topics
+* [Installation](#installation)
+* [Mage](#mage)
+* [Development Setup](#development-setup)
+  * [Installation](#installation)
+  * [Run tests](#run-tests)
+  * [Tests state](#tests-state)
+* [Common information](#common-information)
+  * [MLflow source code](#mlflow-source-code)
+  * [Protos](#protos)
+  * [Request validation](#request-validation)
+  * [Data access](#data-access)
+  * [Linting](#linting)
+* [Usage](#usage)
+  * [Go Server](#go-server)
+  * [Building the Go binary](#building-the-go-binary)
+  * [Client-side Go implementation](#client-side-go-implementation)
+  * [Go store in Python](#go-store-in-python)
+* [Misc](#misc)
+  * [Debug Failing Tests](#debug-failing-tests)
+  * [Supported endpoints](#supported-endpoints)
+  * [Targeting Local Postgres in Python Tests](#targeting-local-postgres-in-python-tests)
+
+## Installation
+
+This package is not yet available on PyPI and currently requires the [Go SDK](https://go.dev) to be installed.
+
+You can then install the package via pip:
+```bash
+pip install git+https://github.com/jgiannuzzi/mlflow-go.git
+```
+
+## Mage
+
+This repository uses [mage](https://magefile.org/) to streamline some utility functions.
+
+```bash
+# Install mage (already done in the dev container)
+go install github.com/magefile/mage@v1.15.0
+
+# See all targets
+mage
+
+# Execute single target
+mage generate
+```
+
+The beauty of Mage is that we can use regular Go code for our scripting.  
+That being said, we are not married to this tool.
+
+## Development Setup
+
+### Installation
+
+```bash
+# Install our Python package and its dependencies
+pip install -e .
+
+# Install the dreaded psycho
+pip install psycopg2-binary
+
+# Archive the MLFlow pre-built UI
+tar -C /usr/local/python/current/lib/python3.8/site-packages/mlflow -czvf ./ui.tgz ./server/js/build
+
+# Clone the MLflow repo
+git clone https://github.com/jgiannuzzi/mlflow.git -b master .mlflow.repo
+
+# Add the UI back to it
+tar -C .mlflow.repo/mlflow -xzvf ./ui.tgz
+
+# Install it in editable mode
+pip install -e .mlflow.repo
+```
+
+or run:
+```bash
+mage configure
```

Comment:
seems like this should be the first, not alternate, option, since so much simpler